examples/dt/nova/nova-three-cells/edpm/nodeset2/kustomization.yaml

```diff
@@ -0,0 +1,34 @@
+---
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+components:
+  - ../../../../../../dt/nova/nova-three-cells/edpm/nodeset
+
+resources:
+  - values.yaml
+
+patches:
+  - target:
+      kind: OpenStackDataPlaneNodeSet
+      name: openstack-edpm
+    patch: |-
+      - op: replace
+        path: /metadata/name
+        value: openstack-edpm-2
+      - op: replace
+        path: /spec/services/13
+        value: nova-cell-2
```

Comment:
This patching feels strange to me. We take default NodeSet and duplicate it for cell2 and patch it to connect the computes in it to cell2. But we 
i) don't define the computes of NodeSet/openstack-edpm-2 here, 
ii) we have a complete NodeSet  parameter definition with the specific compute in nodeset2/values.yaml 

So basically we have two ways to pass values to NodeSet/openstack-edpm2:
* here with the kustomize patch
* via nodeset2/values.yaml that is applied by another kustomize patch I assume.

This make it hard to follow what parameter comes from where. I don't want to block on this but I feel this needs to be refactored if possible to provide parameters to the NodeSet from a single source to help maintaining the kustomization. Is it something that can be done in a follow up? 

@abays how do you feel about this issue? Do you agree that this is a problem? If so do you think it can be refactored to a single source? Is it something new in the proposed DT or is it widespread across the DTs?

Reply:
> I feel this needs to be refactored if possible to provide parameters to the NodeSet from a single source to help maintaining the kustomization.

So what I think might happen when using this in CIFMW is the same problem that was encountered in this PR [1].  Since ultimately the `lib` dataplane nodeset is used [2] by both nodesets, they will collide with each other when CIFMW's `kustomize_deploy` role generates all of the YAMLs ahead time.  But maybe you somehow found a way around the problem.  I'll be curious to see the test results for this.  Since this doesn't change any existing, underlying `lib`, `va`, or `dt` content, I'm good with merging it for now.

[1] https://github.com/openstack-k8s-operators/architecture/pull/378
[2] https://github.com/jamepark4/architecture/blob/191d2bfabdc1355f5ddce7bf47a264fa46c12b45/dt/nova/nova-three-cells/edpm/nodeset/kustomization.yaml#L20

Reply:
Using CIFMW downstream I'm not seeing any collisions since the automation/vars uses two different values: `edpm-nodeset-values` and `edpm-nodeset2-values`.  With nodeset2's name being updated by [1].  `kustomize_deploy` ends up generating two yaml definitions, nodeset.yaml and nodeset2.yaml.  The different Nodeset names allows them to be deployed without any issues.

[1] https://github.com/openstack-k8s-operators/architecture/pull/401/files#diff-20af9459981c1a0588160942c3490649f5cf4fd222526cea08c3244208fb48ecR13

Reply:
That's great news.  @mnietoji maybe you can review how two nodesets were used here and apply the same design to fix your PR (if you haven't already figured it out on your own)?